src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -56,13 +57,22 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner = owner;
                 _subjectBuffer = subjectBuffer;
 
+                // track context changes
```

Comment:
We were never doing this before? Was this scenario always broken?

Reply:
we did this by checking last documentId with what document buffer is associated with at foreground thread when we actually update state. rather than when context changed reported by workspace.

Reply:
This was important due to racey interleavings of events from the diagnostics service and the workspace service.  I'm not convinced this won't just lead to a problem where tagger diagnostics beocme out of sync.